pydantic_ai_slim/pydantic_ai/providers/huggingface.py

```diff
@@ -54,11 +130,55 @@ def model_profile(self, model_name: str) -> ModelProfile | None:
             return None
 
         model_name = model_name.lower()
-        provider, model_name = model_name.split('/', 1)
-        if provider in provider_to_profile:
-            return provider_to_profile[provider](model_name)
+        model_prefix, model_suffix = model_name.split('/', 1)
+
+        base_profile: ModelProfile | None = None
+        if model_prefix in provider_to_profile:
+            base_profile = provider_to_profile[model_prefix](model_suffix)
+
+        # fetch model capabilities
+        router_info = _get_router_info(model_name)
```

Comment:
Unfortunately I think it breaks expectations too much for the sync `Provider.model_profile` method to be doing a HTTP request 😬 We could make it a bit less problematic by moving the request to a thread using `_utils.run_in_executor` but I'm still uncomfortable with this to the point where I can't see myself merging this...

Overriding `self._client` below is also problematic because these providers are often reused across multiple model instances.

Is there any way we could achieve your goals by waiting to do this capabilities checking / provider selection until `Model.request`?